### ISSUE TITLE
bug: IonIcon giving error with vitest

### ISSUE THREAD

**Issue by asif-diebold** [ionic-team/ionic-framework#28779]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

While running vitest on the components using IonIcon, the test is failing with following error though the image is rendered properly in the project
Using IonIcon as: <IonIcon src="/image.svg"/>
<img width="664" alt="error" src="https://github.com/ionic-team/ionic-framework/assets/155605190/74e66ab4-f80b-45ec-8db1-c792371f408a">

It works fine when using ion icon with with inbuilt icons provided by ionic but causes issues when we try to use any image url with it.
 

### Expected Behavior

The IonIcon should work with vitest properly when using image url with it.

### Steps to Reproduce

1. Create any react component in ionic framework.
2. Use IonIcon inside the component as <IonIcon src={imagepath}/> using image path as src in IonIcon.
3. Run Vitest on the component

### Ionic Info

Ionic:

 "@ionic/react": "^7.0.10",
 "@ionic/react-router": "^7.3.1"

vite:

"vite": "4.3.9",
"vite-plugin-dts": "2.3.0",
"vite-plugin-eslint": "^1.8.1",
"vite-plugin-svgr": "^4.2.0",
"vite-tsconfig-paths": "^4.0.2",
"vitest": "0.32.0"

**Comment by rdlabo** [comment]
@asifrahman112  Is the path `/image.svg` correct?
Angular copies everything under `assets` into the build file, so in most cases it should be `/assets/image.svg` or `/assets/**/image.svg`.

**Comment by asifrahman112** [comment]
> @asifrahman112 Is the path `/image.svg` correct? Angular copies everything under `assets` into the build file, so in most cases it should be `/assets/image.svg` or `/assets/**/image.svg`.

@rdlabo yes the path is correct...since its loading correctly in my project and I can access the image on that path. But the test fails with inavlid url message

**Comment by rdlabo** [comment]
@asifrahman112  Thanks reply. Now, try moving your SVG files under `/assets`. The error text in the image says it is looking in `/assets`.

> Failed to parse URL from /assets/nozzle.svg

**Comment by asifrahman112** [comment]
@rdlabo The image is already placed in assets folder in the root public directory

**Comment by rdlabo** [comment]
https://github.com/rdlabo/ionic-react-vite-src-icon/commit/2d48a704a6396765f73bd2ab225c1cc8ba43b618

@asifrahman112  I had mistakenly commented on some of the configurations. Please check the project that we have made which is displayed correctly. This project is based https://github.com/ionic-team/starters/tree/main/react-vite

**Comment by liamdebeasi** [comment]
@asifrahman112 Can you provide a GitHub repo with a minimal reproduction?

**Comment by asifrahman112** [comment]
@rdlabo The image displays correctly in my project. Only the unit test fails on vitest with the error I have mentioned above

**Comment by asifrahman112** [comment]
@liamdebeasi I have added the stackblitz url for the issue reproduction

**Comment by liamdebeasi** [comment]
There's likely an issue with the asset path either not being defined or being defined incorrectly when loading the Ionicon. I'm going to move this over to the Ionicons repo so we can take a closer look.
